dpnp/dpnp_utils/dpnp_utils_linearalgebra.py

```diff
@@ -103,60 +104,59 @@ def _copy_array(x, dep_events, host_events, contig_copy=False, dtype=None):
     return x
 
 
-def _gemm_batch_matmul(exec_q, x1, x2, res, x1_is_2D, x2_is_2D, dev_tasks_list):
-    # If input array is F-contiguous, we need to change the order to C-contiguous.
-    # because mkl::gemm_bacth needs each 2D array to be F-contiguous but
-    # when the input array is F-contiguous, the data of 2D array
-    # that needs to be called in mkl::gemm_batch are not contiguous.
-    ht_tasks_list = []
-    contig_copy = not x1.flags.c_contiguous
-    x1 = _copy_array(x1, dev_tasks_list, ht_tasks_list, contig_copy=contig_copy)
-    contig_copy = not x2.flags.c_contiguous
-    x2 = _copy_array(x2, dev_tasks_list, ht_tasks_list, contig_copy=contig_copy)
-
-    x1_strides = x1.strides
-    x2_strides = x2.strides
-    res_strides = res.strides
-
-    # need to standardize to use in ti._contract_iter2
-    x1_strides = _standardize_strides(x1_strides, x1_is_2D, x1.shape, x1.ndim)
-    x2_strides = _standardize_strides(x2_strides, x2_is_2D, x2.shape, x2.ndim)
-
-    batch_size = res.shape[:-2][0]
-    stridea = x1_strides[0]
-    strideb = x2_strides[0]
-    stridec = res_strides[-3]
-
-    if x1.ndim > 3:
-        iter = ti._contract_iter2(
-            res.shape[:-2], x1_strides[:-2], x2_strides[:-2]
-        )
+def _define_contig_flag(x):
+    """
+    Determines if the data in last two dimensions of array `x` are
+    c_contiguous or f_contiguous. For 2D arrays, it is the same as using
+    x.flags.c_contiguous or x.flags.f_contiguous.
+    """
+
+    flag = False
+    x_strides = x.strides
+    x_shape = x.shape
+    if x.ndim == 1:
+        return True
+    else:
+        x_is_c_contiguous = x_strides[-1] == 1 and x_strides[-2] == x_shape[-1]
+        x_is_f_contiguous = x_strides[-2] == 1 and x_strides[-1] == x_shape[-2]
+        if x_is_c_contiguous or x_is_f_contiguous:
+            flag = True
+        return flag
```

Comment:
Such `else` block produces pylint warning, since in the above `if` was `return`
```suggestion

    x_is_c_contiguous = x_strides[-1] == 1 and x_strides[-2] == x_shape[-1]
    x_is_f_contiguous = x_strides[-2] == 1 and x_strides[-1] == x_shape[-2]
    if x_is_c_contiguous or x_is_f_contiguous:
        return True
    return False
```